manager/controllers/default/resource/update.class.php

```diff
@@ -114,21 +123,23 @@ public function process(array $scriptProperties = array()) {
         /* register FC rules */
         $this->resourceArray = $this->resource->toArray();
         $overridden = $this->checkFormCustomizationRules($this->resource);
-        $this->resourceArray = array_merge($this->resourceArray,$overridden);
-
-        $this->resourceArray['published'] = intval($this->resourceArray['published']) == 1 ? true : false;
-        $this->resourceArray['hidemenu'] = intval($this->resourceArray['hidemenu']) == 1 ? true : false;
-        $this->resourceArray['isfolder'] = intval($this->resourceArray['isfolder']) == 1 ? true : false;
-        $this->resourceArray['richtext'] = intval($this->resourceArray['richtext']) == 1 ? true : false;
-        $this->resourceArray['searchable'] = intval($this->resourceArray['searchable']) == 1 ? true : false;
-        $this->resourceArray['cacheable'] = intval($this->resourceArray['cacheable']) == 1 ? true : false;
-        $this->resourceArray['deleted'] = intval($this->resourceArray['deleted']) == 1 ? true : false;
-        $this->resourceArray['uri_override'] = intval($this->resourceArray['uri_override']) == 1 ? true : false;
+        $this->resourceArray = array_merge($this->resourceArray, $overridden);
+
+        $this->resourceArray['published'] = (int)$this->resourceArray['published'] == 1;
```

Comment:
I am not sure what would be executed first in that code: the int casting or the comparison. So i would prefer:
`$this->resourceArray['syncsite'] = intval($this->resourceArray['xxx']) == 1`

Reply:
What do you mean? Of course int casting! Besides `(int)` is up to 50% faster then `intval()`.

Reply:
I would go for a stricter check but casting to int this way looks good to me.